checker/src/org/checkerframework/checker/index/searchindex/SearchIndexTransfer.java

```diff
@@ -0,0 +1,106 @@
+package org.checkerframework.checker.index.searchindex;
+
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.checker.index.IndexAbstractTransfer;
+import org.checkerframework.checker.index.IndexUtil;
+import org.checkerframework.checker.index.qual.NegativeIndexFor;
+import org.checkerframework.checker.index.qual.SearchIndexFor;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+
+/**
+ * The transfer function for the SearchIndexFor checker. Allows {@link SearchIndexFor} to be refined
+ * to {@link NegativeIndexFor}.
+ */
+public class SearchIndexTransfer extends IndexAbstractTransfer {
+
+    // The ATF (Annotated Type Factory).
+    private SearchIndexAnnotatedTypeFactory aTypeFactory;
+
+    private CFAnalysis analysis;
+
+    public SearchIndexTransfer(CFAnalysis analysis) {
+        super(analysis);
+        this.analysis = analysis;
+        aTypeFactory = (SearchIndexAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * {@link org.checkerframework.checker.index.qual.SearchIndexFor {@code @SearchIndexFor}} is
+     * refined only when it is compared to these values.
+     */
+    private enum ValidComparisons {
+        NEGATIVE_ONE,
+        ZERO
+    }
+
+    /**
+     * If the left value is exactly the value of {@code toCompareTo}, and the right side has type
+     * {@link SearchIndexFor}, then the right side's new value in the store should become {@link
+     * NegativeIndexFor}.
+     *
+     * <p>For example, this allows the following code to typecheck:
+     *
+     * <pre>{@code
+     * @SearchIndexFor("a") int index = Arrays.binarySearch(a, y);
+     * if (index < 0) {
+     *     @NegativeIndexFor("a") int negInsertionPoint = index;
+     * }
+     * }</pre>
+     */
+    private void specialHandlingForBinarySearch(
```

Comment:
This method name doesn't convey what the routine does (which has nothing to do with binarySearch, though it is related to the `@SearchIndex` annotation and to refinement).  The documentation also doesn't say anything about greater-than or less-than tests (except indirectly in the example), which is relevant to the specification.